src/views/TechnicalAssistance/AdviceLetterForm/RecommendationsForm.tsx

```diff
@@ -0,0 +1,262 @@
+          {
+            text: t(`Request ${trbRequestId}`),
+            url: `/trb/${trbRequestId}/advice`
+          },
+          {
+            text: t('adviceLetterForm.heading'),
+            url: `/trb/${trbRequestId}/advice/recommendations`
+          },
+          {
+            text: t('adviceLetterForm.addRecommendation')
+          }
+        ]}
+      />
+
+      {
+        /* Error alert for gql errors */
+        showFormError && (
+          <Alert type="error" className="margin-bottom-5" slim>
+            {t('adviceLetterForm.error', {
+              action: 'saving',
+              type: 'recommendation'
+            })}
+          </Alert>
+        )
+      }
+
+      <h1 className="margin-bottom-0">
+        {t('adviceLetterForm.addRecommendation')}
+      </h1>
+      {/* Required fields text */}
+      <HelpText className="margin-top-1 margin-bottom-2 text-base">
+        <Trans
+          i18nKey="technicalAssistance:requiredFields"
+          components={{ red: <span className="text-red" /> }}
+        />
+      </HelpText>
+
+      <Form onSubmit={e => e.preventDefault()} className="maxw-tablet">
+        {/* Title */}
+        <Controller
+          name="title"
+          control={control}
+          render={({ field, fieldState: { error } }) => {
+            return (
+              <FormGroup className="margin-top-3" error={!!error}>
+                <Label className="text-normal" htmlFor="title" required>
+                  {t('Title')}
+                </Label>
+                {error && <ErrorMessage>{t('fillBlank')}</ErrorMessage>}
+                <TextInput
+                  type="text"
+                  id="title"
+                  {...field}
+                  ref={null}
+                  required
+                />
+              </FormGroup>
+            );
+          }}
+        />
+
+        {/* Description */}
+        <Controller
+          name="recommendation"
+          control={control}
+          render={({ field, fieldState: { error } }) => {
+            return (
+              <FormGroup className="margin-top-3" error={!!error}>
+                <Label
+                  className="text-normal"
+                  htmlFor="recommendation"
+                  required
+                >
+                  {t('Description')}
+                </Label>
+                {error && <ErrorMessage>{t('errors.fillBlank')}</ErrorMessage>}
+                <TextAreaField
+                  id="recommendation"
+                  {...field}
+                  ref={null}
+                  required
+                />
+              </FormGroup>
+            );
+          }}
+        />
+
+        {/* Links */}
+        <LinksArrayField />
+
+        {/* Save */}
+        <Button
+          type="submit"
+          className="margin-top-6"
+          disabled={
+            watch('title').length === 0 ||
+            watch('recommendation').length === 0 ||
+            isSubmitting
+          }
```

Comment:
I like how you handled this and is totally fine, just wanted to give an idea for something to scale with lots of fields.  Honestly not sure why RHF doesn't have something cleaner OOTB for this, not even sure if my recommendation works for all cases (conditionally required fields etc, but it works if all fields are required).  I want to look more into this.  No need to change this! 

```
disabled={
    Object.keys(dirtyFields).length !==
      Object.keys(defaultValues).length || isSubmitting
  }
```

Reply:
If it checks against the default/initial value, then if the value is reverted/erased, it would no longer be dirty.  I tested with with empty string and it seemed to work as expected.